app/code/community/Ebizmarts/MailChimp/Model/ClearEcommerce.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+/**
+ * MailChimp For Magento
+ *
+ * @category  Ebizmarts_MailChimp
+ * @author    Ebizmarts Team <info@ebizmarts.com>
+ * @copyright Ebizmarts (http://ebizmarts.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ * @date:     7/24/19 2:47 PM
+ * @file:     ClearEcommerce.php
+ */
+class Ebizmarts_MailChimp_Model_ClearEcommerce
+{
+    /**
+     * @var Ebizmarts_MailChimp_Helper_Data
+     */
+    protected $_helper;
+
+    /**
+     * @var Ebizmarts_MailChimp_Helper_Date
+     */
+    protected $_dateHelper;
+
+    public function __construct()
+    {
+        $this->_helper = Mage::helper('mailchimp');
+        $this->_dateHelper = Mage::helper('mailchimp/date');
+    }
+
+    /**
+     * @return Ebizmarts_MailChimp_Helper_Data|Mage_Core_Helper_Abstract
+     */
+    protected function getHelper()
+    {
+        return $this->_helper;
+    }
+
+    /**
+     * @return Ebizmarts_MailChimp_Helper_Date|Mage_Core_Helper_Abstract
+     */
+    protected function getDateHelper()
+    {
+        return $this->_dateHelper;
+    }
+
+    /**
+     * Process all types of data from eCommerce data to delete
+     * non active products, quotes, customers, etc. from the table.
+     */
+    public function cleanEcommerceData()
+    {
+        $this->processData($this->getData('QUO'), 'QUO');
+        $this->processData($this->getData('PRO'), 'PRO');
+        $this->processData($this->getData('CUS'), 'CUS');
+        $this->processData($this->getData('PRL'), 'PRL');
+        $this->processData($this->getData('PCD'), 'PCD');
+    }
+
+    /**
+     * @param $data
+     * @param $type
+     */
+    protected function processData($data, $type)
+    {
+        $ids = array();
+        foreach ($data as $item) {
+            $ids []= $item->getId();
+        }
+
+        $reverseIds = $this->processReverseData($type);
+        $ids = array_merge($ids, $reverseIds);
+
+        if (!empty($ids)) {
+            $this->deleteEcommerceRows($ids, $type);
+        }
+    }
+
+    protected function processReverseData($type)
+    {
+        $ids = array();
+        $eData = $this->getEcommerceRows($type);
+
+        foreach ($eData as $eItem) {
+            $ids []= $eItem['related_id'];
+        }
+
+        return $ids;
+    }
+
+    /**
+     * @param $type
+     * @param bool $filter
+     * @return array
+     * @throws Mage_Core_Model_Store_Exception
+     */
+    protected function getData($type, $filter = true)
```

Comment:
Rename to something that tells more about what it does.
For example: getItemsToDelete